code/game/objects/structures/window.dm

```diff
@@ -78,6 +80,40 @@
 	if (flags_1 & ON_BORDER_1)
 		AddElement(/datum/element/connect_loc, loc_connections)
 
+/obj/structure/window/mouse_drop_receive(atom/dropping, mob/user, params)
+	. = ..()
+	if (added_leaning)
+		return
+	/// For performance reasons and to cut down on init times we are "lazy-loading" the leaning component when someone drags their sprite onto us, and then calling dragging code again to trigger the component
+	/// WALLSTATION EDIT BEGIN
+	var/offset = 11
+	if(!fulltile)
```

Comment:
purely a personal one but i think it would be better just make it return here because you would always be clipping through a directional window anyway

Reply:
Yeah I'll probably do that instead